### PR TITLE
Removes all wrapped errors in k8s.go

### PR DESCRIPTION
**Pull request by epps**
# Description

Removes all wrapped errors in k8s.go.

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/k8s.go [hunk 1]
```diff
@@ -208,7 +209,7 @@ func k8sOfflineStoreFactory(config pc.SerializedConfig) (Provider, error) {
 
 	serializedBlob, err := k8.StoreConfig.Serialize()
 	if err != nil {
-		return nil, fmt.Errorf("could not serialize blob store config")
+		return nil, err
```
Comment (anthonylasso) [inline provider/k8s.go:212]: do we need an error type here? if not ignore

**Review by anthonylasso — APPROVED** [review]

File: provider/k8s.go (hunk 1 above)
Comment (sdreyer) [reply provider/k8s.go:212]: Nah its handled by another function

File: provider/k8s.go [hunk 2]
```diff
@@ -271,19 +272,19 @@ func (config *LocalExecutorConfig) Serialize() ([]byte, error) {
 func (config *LocalExecutorConfig) Deserialize(data []byte) error {
 	err := json.Unmarshal(data, config)
 	if err != nil {
-		return fmt.Errorf("deserialize executor config: %w", err)
+		return err
```
Comment (sdreyer) [inline provider/k8s.go:275]: Need an internal error here

File: provider/k8s.go [hunk 3]
```diff
@@ -271,19 +272,19 @@ func (config *LocalExecutorConfig) Serialize() ([]byte, error) {
...
 	}
 	return nil
 }
 
 func NewLocalExecutor(config Config, logger *zap.SugaredLogger) (Executor, error) {
 	localConfig := LocalExecutorConfig{}
 	if err := localConfig.Deserialize(config); err != nil {
-		return nil, fmt.Errorf("failed to deserialize config")
+		return nil, fferr.NewInternalError(err)
```
Comment (sdreyer) [inline provider/k8s.go:283]: Throw this internal error in the Deserialize function instead

File: provider/k8s.go [hunk 4]
```diff
@@ -300,7 +301,7 @@ type KubernetesExecutor struct {
 func (kube *KubernetesExecutor) isDefaultImage() (bool, error) {
 	parse, err := dp.Parse(kube.image)
 	if err != nil {
-		return false, fmt.Errorf("invalid image name: %w", err)
+		return false, fferr.NewInvalidArgument(err)
```
Comment (sdreyer) [inline provider/k8s.go:304]: We may want to prepend "invalid Docker image name: err" to this, unsure what error that library is going to throw

File: provider/k8s.go [hunk 5]
```diff
@@ -839,7 +840,7 @@ func (k8s *K8sOfflineStore) sqlTransformation(config TransformationConfig, isUpd
 func (k8s *K8sOfflineStore) checkArgs(args metadata.TransformationArgs) (metadata.KubernetesArgs, error) {
 	k8sArgs, ok := args.(metadata.KubernetesArgs)
 	if !ok {
-		return metadata.KubernetesArgs{}, fmt.Errorf("invalid type used for Kubernetes Arguments")
+		return metadata.KubernetesArgs{}, fferr.NewInvalidArgument(fmt.Errorf("args is not of type KubernetesArgs"))
```
Comment (sdreyer) [inline provider/k8s.go:843]: Think this should be an internal error since if the struct is wrong then its our fault

**Review by sdreyer — APPROVED** [review]

**Pull request merged**